File: cucumber_java/first_taste/08/step_definitions/CheckoutSteps.java
/***
 * Excerpted from "The Cucumber for Java Book",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/srjcuc for more book information.
***/
package step_definitions;

import cucumber.api.java.en.*;
import cucumber.api.PendingException;
import implementation.Checkout;

public class CheckoutSteps {
    @Given("^the price of a \"(.*?)\" is (\\d+)c$")
    public void thePriceOfAIsC(String name, int price) throws Throwable {
        int bananaPrice = price;
    }

    @When("^I checkout (\\d+) \"(.*?)\"$")
    public void iCheckout(int itemCount, String itemName) throws Throwable {
        Checkout checkout = new Checkout();
        checkout.add(itemCount, bananaPrice);
    }

    @Then("^the total price should be (\\d+)c$")
    public void theTotalPriceShouldBeC(int arg1) throws Throwable {
        // Write code here that turns the phrase above into concrete actions
        throw new PendingException();
    }
}

File: cucumber_java/appendix/scala/01/src/test/scala/checkout/CheckoutSteps.scala
/***
 * Excerpted from "The Cucumber for Java Book",
 * published by The Pragmatic Bookshelf.
 * Copyrights apply to this code. It may not be used to create training material, 
 * courses, books, articles, and the like. Contact us if you are in doubt.
 * We make no guarantees that this code is fit for any purpose. 
 * Visit http://www.pragmaticprogrammer.com/titles/srjcuc for more book information.
***/
package checkout

import cucumber.api.scala.{ScalaDsl, EN}
import org.junit.Assert._

class CheckoutSteps extends ScalaDsl with EN {

  When("""^I scan a "(.*?)"$"""){ (itemName:String) =>
    CheckoutWorld.checkout.scan(itemName)
  }
  
  Then("""^the total is (\d+)c$"""){ (expectedTotal:Int) =>
    assertEquals(expectedTotal, CheckoutWorld.checkout.getTotal)
  }
}